pkg/operator/ceph/object/rgw.go

```diff
@@ -211,13 +205,20 @@ func (c *clusterConfig) startRGWPods() error {
 			depIDToRemove := currentRgwInstances - 1
 			depNameToRemove := fmt.Sprintf("%s-%s-%s", AppName, c.store.Name, k8sutil.IndexToName(depIDToRemove))
 			if err := k8sutil.DeleteDeployment(c.context.Clientset, c.store.Namespace, depNameToRemove); err != nil {
-				logger.Warning("error during deletion of deployment %q resource. %v", depNameToRemove, err)
+				logger.Warningf("error during deletion of deployment %q resource. %v", depNameToRemove, err)
 			}
 			currentRgwInstances = currentRgwInstances - 1
 			i++
 
+			// Delete the Secret key
+			secretToRemove := fmt.Sprintf("%s-%s-%s-keyring", AppName, c.store.Name, k8sutil.IndexToName(depIDToRemove))
```

Comment:
Can there be a common method for creating this secret name when it is created and deleted? It's not a simple name.

Reply:
Can do!